packages/pluggableWidgets/video-player-native/src/VideoPlayer.tsx

```diff
@@ -14,65 +13,49 @@ const enum StatusEnum {
     NOT_READY = "not-ready"
 }
 
-interface State {
-    aspectRatio?: number;
-    status: StatusEnum;
-}
-
-export class VideoPlayer extends Component<Props, State> {
-    readonly state: State = {
-        status: StatusEnum.NOT_READY
-    };
-
-    private readonly onLoadStartHandler = this.onLoadStart.bind(this);
-    private readonly onLoadHandler = this.onLoad.bind(this);
-    private readonly onErrorHandler = this.onError.bind(this);
-    private readonly styles = flattenStyles(defaultVideoStyle, this.props.style);
-
-    render(): JSX.Element {
-        const uri = this.props.videoUrl && this.props.videoUrl.value;
-
-        const styles = { ...this.styles.container };
-
-        if (this.props.aspectRatio && this.state.aspectRatio) {
-            styles.aspectRatio = this.state.aspectRatio;
+export function VideoPlayer(props: VideoPlayerProps<VideoStyle>): ReactElement {
+    const [styles, setStyles] = useState(flattenStyles(defaultVideoStyle, props.style));
+    const [status, setStatus] = useState(StatusEnum.NOT_READY);
+    const [videoAspectRatio, setVideoAspectRatio] = useState(0);
+
+    useEffect(() => {
+        const alteredStyles = JSON.parse(JSON.stringify(styles));
```

Comment:
flattenStyles returns an object, why this parse and stringify is necessary here?

Reply:
Since we are changing the second child of the object, we need to deep copy otherwise it throws an error

Reply:
Done